### ISSUE TITLE
update use of jazzy to fix Middleware docs rendering

### ISSUE THREAD

**Issue by DivineDominion** [ReSwift/ReSwift#437]
Have a look at the docs on Middleware: http://reswift.github.io/ReSwift/master/Other%20Typealiases.html#/s:7ReSwift10Middlewarea

It reads

    public typealias Middleware<State> = (@escaping DispatchFunction, @escaping () -> State?)

But the code is

    public typealias Middleware<State> = (@escaping DispatchFunction, @escaping () -> State?)
        -> (@escaping DispatchFunction) -> DispatchFunction

As a quick-fix, we could put the definition on one line and silence SwiftLint. 

I tried to run the latest `jazzy` v0.13.0 locally, but the result was the same. I files realm/jazzy#1148

**Comment by DivineDominion** [comment]
While I look at it, I personally would prefer if the signature gave away the expected closure argument names:

```
public typealias Middleware<State> = (_ dispatch: @escaping DispatchFunction, _ getState: @escaping () -> State?) -> (_ next: @escaping DispatchFunction) -> DispatchFunction
```

**Comment by DivineDominion** [comment]
realm/jazzy#1148 is fixed now in v0.13.1 so the documentation should work.

Now it's just that ReSwift uses @agentk's jazzy fork which we have to replace in the `gh-pages` branch.

**Comment by DivineDominion** [comment]
Thanks to @mjarvis the docs do auto-regenerate now on GH Actions
